src/MediaBundleForm.php

```diff
@@ -101,6 +139,28 @@ public function form(array $form, FormStateInterface $form_state) {
   /**
    * {@inheritdoc}
    */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+
+    // Let the selected plugin validate its settings.
+    $plugin = $this->entity->getType()->getPluginId();
```

Comment:
Plugin could have been changed at the submit. In that case this line won't get correct plugin ID any more. We should read from submitted values instead.